argmin/src/solver/powell/mod.rs

```diff
@@ -0,0 +1,97 @@
+use crate::core::{
+    ArgminFloat, CostFunction, DeserializeOwnedAlias, Executor, IterState, LineSearch,
+    OptimizationResult, SerializeAlias, Solver, State,
+};
+use argmin_math::{ArgminAdd, ArgminDot, ArgminSub, ArgminZeroLike};
+#[cfg(feature = "serde1")]
+use serde::{Deserialize, Serialize};
+use std::mem;
+
+#[derive(Clone)]
+#[cfg_attr(feature = "serde1", derive(Serialize, Deserialize))]
+pub struct PowellLineSearch<P, L> {
+    search_vectors: Vec<P>,
+    linesearch: L,
+}
+
+impl<P, L> PowellLineSearch<P, L> {
+    pub fn new(initial_search_vectors: Vec<P>, linesearch: L) -> Self {
+        PowellLineSearch {
+            search_vectors: initial_search_vectors,
+            linesearch,
+        }
+    }
+}
+
+impl<O, P, F, L> Solver<O, IterState<P, (), (), (), F>> for PowellLineSearch<P, L>
+where
+    O: CostFunction<Param = P, Output = F>,
+    P: Clone
+        + SerializeAlias
+        + DeserializeOwnedAlias
+        + ArgminAdd<P, P>
+        + ArgminZeroLike
+        + ArgminSub<P, P>
+        + ArgminDot<P, F>,
+    F: ArgminFloat,
+    L: Clone + LineSearch<P, F> + Solver<O, IterState<P, (), (), (), F>>,
+{
+    const NAME: &'static str = "Powell-LS";
+
+    fn next_iter(
+        &mut self,
+        problem: &mut crate::core::Problem<O>,
+        mut state: IterState<P, (), (), (), F>,
+    ) -> Result<(IterState<P, (), (), (), F>, Option<crate::core::KV>), anyhow::Error> {
+        let param = state
+            .take_param()
+            .ok_or_else(argmin_error_closure!(NotInitialized, "not initialized"))?; // TODO add Error message
+
+        // new displacement vector created from displacement vectors of line searches
+        let new_displacement = param.zero_like();
+        let mut best_direction: (usize, F) = (0, float!(0.0));
+
+        // init line search
+        let (ls_state, _) = self.linesearch.init(problem, state.clone())?;
```

Comment:
What is the reason for initializing the line search here explicitly? This is typically called by the `Executor`.